eng/pipelines/templates/stages/archetype-sdk-tests.yml

```diff
@@ -117,6 +117,10 @@ stages:
         jobs:
         - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
           parameters:
+            SparseCheckoutPaths:
```

Comment:
No null checking on `${{ parameters.ServiceDirectory }}`
The script will fail when no service directory passed in. @benbp is this valid case?

Reply:
Base on my testing, if no service directory pass in, then it does not fail at the this step, but failed at next step as we do not actually checkout the useful files.
The value we pass is `sdk/not-specified/**/*.json`.
IMO, we have to do if/else checking.

Reply:
What are the cases where service directory is empty?

Reply:
I don't have any cases. Do we want to remove the if/else check?

Reply:
@benbp and @sima-zhu I don't believe this change is correct place to make this change. This change should be made [here](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml#L63) and applied only when the servicedirectory is passed in. The parameter you're overriding here isn't correct. The SparseCheckoutPaths parameter is something that's supposed to be passed through, not something that's supposed to be set here.

Reply:
Talked offline. This is the place to address this in short run. In long run, recording files move to other repo, then the exclusion is not needed